src/disk_interface.cc

```diff
@@ -132,11 +133,11 @@ bool StatAllFilesInDir(const string& dir, map<string, TimeStamp>* stamps,
 // DiskInterface ---------------------------------------------------------------
 
 bool DiskInterface::MakeDirs(const string& path) {
-  string dir = DirName(path);
+  const std::string& dir = DirName(path);
```

Comment:
`DirName` doesn't return a const &, therefore this won't avoid the copy (it actually introduces an indirection when optimizations are turned off). The only way to avoid the copy would be to use `std::string_view`, but we'll have to wait for C++17 for that.

Reply:
@jhasse Yes, you are totally right!

Reply:
Why can't StringPiece be used in place of std::string_view?